src/main/java/ninja/eivind/hotsreplayuploader/concurrent/tasks/UploadTask.java

```diff
@@ -53,9 +53,12 @@ protected ReplayFile call() throws Exception {
                 preStatus = provider.getPreStatus(replay);
             } catch (MpqException e) {
                 LOG.warn("Could not parse replay, deferring to upload." + take, e);
-                preStatus = Status.NEW;
+
+                preStatus = Status.EXCEPTION;
             }
-            if (preStatus == Status.UPLOADED || preStatus == Status.UNSUPPORTED_GAME_MODE) {
+            if (preStatus == Status.UPLOADED ||
+                preStatus == Status.UNSUPPORTED_GAME_MODE ||
+                preStatus == Status.EXCEPTION) {
```

Comment:
Aye. We don't want to discard replays that @barrett777 might be able to parse in case the protocol changes and this app is abandoned. That would break it permanently.